### PR TITLE
select_index extended

### PR DESCRIPTION
**Pull request by samwaseda**

### CODE REVIEW

**Comment by jan-janssen** [comment]
There again seems to be a Python 2.7 issue at least on windows: 

```
======================================================================
857ERROR: test_repeat (atomistics.structure.test_atoms.TestAtoms)
858----------------------------------------------------------------------
859Traceback (most recent call last):
860  File "C:\projects\pyiron\tests\atomistics\structure\test_atoms.py", line 371, in test_repeat
861    basis.spin[ind] = 1
862  File "pyiron\atomistics\structure\sparse_list.py", line 195, in __setitem__
863    if max(key) > self._length:
864TypeError: 'long' object is not iterable
865
866----------------------------------------------------------------------
867Ran 356 tests in 97.251s
868
869FAILED (errors=1, skipped=4)
870
```

File: pyiron/atomistics/structure/atoms.py [hunk 1]
```diff
@@ -676,16 +676,21 @@ def select_index(self, el):
...
             numpy.ndarray: An array of indices of the atoms of the given element
 
         """
         if isinstance(el, str):
-            return np.array([i for i, e in enumerate(self.get_chemical_symbols()) if e == el], dtype=int)
+            return np.where(self.get_chemical_symbols()==el)[0]
         elif isinstance(el, ChemicalElement):
-            return np.array([i for i, e in enumerate(self.get_chemical_elements()) if e == el], dtype=int)
+            return np.where([e==el for e in self.get_chemical_elements()])[0]
+        if isinstance(el, list):
```
Comment (jan-janssen) [inline pyiron/atomistics/structure/atoms.py]: Maybe use `if isinstance(el, (list, np.ndarray))` to be more generic.

File: tests/atomistics/structure/test_atoms.py [hunk 2]
```diff
@@ -667,8 +667,23 @@ def test_occupy_lattice(self):
...
-        self.assertTrue(np.array_equal(self.CO2.select_index("O"), [1, 2]))
+        basis = Atoms(symbols=['Fe', 'Cu', 'Ni', 'Al'], positions=np.random.random((4, 3)), cell=np.eye(3))
+        self.assertTrue(np.array_equal(basis.select_index("Fe"), [0]))
+        self.assertTrue(np.array_equal(basis.select_index("Ni"), [2]))
+        self.assertTrue(np.array_equal(basis.select_index(['Cu', 'Al']), [1, 3]))
+        Fe = basis.convert_element('Fe')
+        Ni = basis.convert_element('Ni')
+        self.assertTrue(np.array_equal(basis.select_index([Fe, Ni]), [0, 2]))
+        pse = PeriodicTable()
+        pse.add_element("Ni", "Ni_up", spin="up")
```
Comment (jan-janssen) [inline tests/atomistics/structure/test_atoms.py]: Setting the spin as a string is possible but confusing, as in practise we commonly expect a float value or a vector.

**Comment by jan-janssen** [comment]
Seems like we reached 70% coverage again! 👍

**Comment by samwaseda** [comment]
I guess erasing `_from_hdf_old` will help a lot

**Comment by jan-janssen** [comment]
It is in there for more than a year, so from my perspective we can remove it:
https://github.com/pyiron/pyiron/commit/5cbcd98d447f6b8e27d686396130852fe25d43a3

**Comment by jan-janssen** [comment]
And seems to be even older the more I dig into it 
https://gitlab.mpcdf.mpg.de/pyiron/pyiron/commit/e17ad22baf3d61486b88d0ae27227645ca2d6555

**Comment by sudarsan-surendralal** [comment]
`_from_hdf_old` is for backward compatibility. If you remove this, some of the old jobs won't load. This will be a problem for early pyiron users (like me :()

**Comment by jan-janssen** [comment]
Old in like before July 2017? Can not we convert these jobs?

**Pull request merged**